Modules/_threadmodule.c

```diff
@@ -80,13 +132,17 @@ ThreadHandle_dealloc(ThreadHandleObject *self)
     }
     HEAD_UNLOCK(&_PyRuntime);
 
-    if (self->joinable) {
-        int ret = PyThread_detach_thread(self->handle);
-        if (ret) {
+    if (self->state == THREAD_HANDLE_RUNNING) {
```

Comment:
comments in this function as to why it is safe to not use the get/set_ functions for atomic access would be nice.

In general i believe this should never be possible to call if another thread had a reference and were calling another method that accessed them?

Reply:
> In general i believe this should never be possible to call if another thread had a reference

Yeah, this is the destructor so nothing else should have a reference and the refcount going to zero is a "syncronizes-with" event, so any change from another thread will be visible here.